### DISCUSSION TITLE
Docker example for Windows - What do I put for "${PWD}" ?

### DISCUSSION THREAD

**Discussion by Nikanoru** [aedocw/epub2tts#169] [Q&A] [answered]
I am pretty much a noob, I followed the readme, but where it says to use the command
 
docker run -v ${PWD}/.local/share/tts:/root/.local/share/tts -v ${PWD}:/root -w /root ghcr.io/aedocw/epub2tts:release mybook.epub --scan

I always get the error: 

docker: Error response from daemon: create ${PWD}/.local/share/tts: "${PWD}/.local/share/tts" includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed. If you intended to pass a host directory, use absolute path.
See 'docker run --help'.

How exactly do I need to change this command to make it work? I tried something like: 

docker run -v C:\Users\Administrator/.local/share/tts:/root/.local/share/tts -v C:\Users\Administrator\:/root -w /root ghcr.io/aedocw/epub2tts:release an.epub --scan

but that was just a guess of mine and does not work.

Any help would be wonderful.

**Comment by aedocw** [comment]
Interesting, my understanding is that powershell is supposed to expand that automatically to whatever your current directory is. I don't use windows very often but if I get a chance in the next day or two I'll try again and see if I can make sense of why the instructions did not work for you.

Hopefully someone else who runs this can help (@danielw97 or @winfran maybe?)

**Comment by danielw97** [comment, marked as answer]
Hi,
Unfortunately I'm not running with docker, although just to double check that you're using powershell and not cmd?
I believe that ${PWD} should work on windows when using powershell as well, so not sure why it wouldn't be for you.\

**Comment by Nikanoru** [comment]
You guys don't know how relieving it is that someone is answering me and trying to help :) 

I was using cmd...>.< I did not know that this command only works in powershell. Now that I reread the readme, it even clearly states to run it in powershell. Sorry for my mistake.

I just had a different error after that saying "FileNotFoundError: [Errno 2] No such file or directory:" but then I had the idea to do a 
"cd C:\Users\Administrator\.local\share\tts" and run the command again from there and now it worked. I am sure experienced users will know that they need to run the docker command from the directory where the epub is sitting, but I did not ^^ 

Maybe the part to "cd" into the directory where the file is one wants to convert with the tts could be added to the readme for noobs like me.

Thanks again for the help guys :)

**Comment by aedocw** [comment]
Really glad we were able to get you going! Thanks for the feedback too, I'll double-check a few times to see why you had to change to that directory. In theory it should have worked from basically whatever starting/working directory you want, but I appreciate you sharing the steps you had to take so I can check on them, and others who might have similar problems will benefit from this as well. Cheers!
